File: aquas/lib/core/legal/legal_documents.dart
// lib/core/legal/legal_documents.dart
//
// IMPORTANT: this is placeholder legal copy written to be reasonable and
// clearly structured, NOT a substitute for review by an actual lawyer.
// Because this app tracks employee location, attendance, and pay data,
// have these reviewed (and update the placeholders marked [ ]) before
// this goes in front of real workers.

/// A single legal document plus the metadata needed to (a) render it and
/// (b) record a legally meaningful acceptance of it.
class LegalDocument {
  final String key; // stable identifier, sent to the backend - never rename
  final String title;
  final String version;
  final String effectiveDate;
  final String content;

  const LegalDocument({
    required this.key,
    required this.title,
    required this.version,
    required this.effectiveDate,
    required this.content,
  });
}

class LegalDocuments {
  const LegalDocuments._();

  static const String lastUpdated = '[Month Year]';
  static const String effectiveDate = '[Month Day, Year]';

  /// Bump this whenever ANY document below changes in a way that's
  /// materially different from what a worker previously accepted (not for
  /// typo fixes). Bumping this is what triggers the mandatory
  /// re-acceptance flow on next login - see LegalAcceptanceService.
  static const String currentAcceptanceVersion = '1.0.0';

  // ---------------------------------------------------------------------
  static const String termsOfUse = '''
TERMS OF USE
Last updated: $lastUpdated

1. WHO THIS APP IS FOR
This app ("the App") is a companion app for CrewControl (also referred to
as "Crewio"), a workforce management platform. The App only works for
workers whose employer ("your Organization") is already a registered
CrewControl customer using it to record attendance, assignments, and pay.
If your Organization does not use CrewControl, this App has no function
for you - there is no independent account you can create outside of an
Organization's CrewControl workspace.

2. YOUR ACCOUNT
- Your account is created and managed by your Organization (typically by
  an office administrator), not by you directly.
- You are responsible for keeping your login credentials confidential and
  for all activity recorded under your account.
- You must use your own account only. Sharing your login, or checking in
  or recording attendance on behalf of another worker, is not permitted
  and may be treated as a violation of your Organization's policies.

3. WHAT YOU CAN DO WITH THE APP
You may use the App to: view your work assignments, record attendance
(check-in/check-out), view salary slips and advances, view and respond to
messages from your office, and receive notifications about the above.

4. WHAT YOU MAY NOT DO
- Attempt to falsify attendance, location, or work records.
- Attempt to access another worker's account or data.
- Reverse-engineer, decompile, or interfere with the App's normal
  operation.
- Use the App for any purpose other than the workforce management
  functions it's built for.

5. DEVICE PERMISSIONS
The App requests location and notification permissions to do its job (see
the Privacy Policy for exactly how each is used). You can decline or later
revoke these permissions in your device settings, but some features (for
example, location-based attendance verification) may stop working if you
do.

6. SUSPENSION AND TERMINATION
Your access can be suspended or ended by your Organization (for example,
if your employment ends) or by CrewControl if we reasonably believe this
App is being misused. Ending your access does not remove records your
Organization is required to keep for payroll, tax, or labor-law purposes.

7. CHANGES TO THESE TERMS
We may update these Terms of Use from time to time. Continuing to use the
App after an update means you accept the revised terms. Material changes
will be flagged the next time you log in.

8. CONTACT
Questions about these Terms of Use should be directed to your
Organization's office administrator, or to [support contact / email].
''';

  // ---------------------------------------------------------------------
  static const String termsAndConditions = '''
TERMS AND CONDITIONS
Last updated: $lastUpdated

1. THE RELATIONSHIP BETWEEN YOU, YOUR ORGANIZATION, AND CREWCONTROL
CrewControl provides workforce management software to your Organization
(your employer or labor contractor). Your Organization is our customer;
you, as a worker, are a user of the software on your Organization's
behalf. Your Organization is responsible for how it uses your data within
the platform (for example, deciding your pay calculations, site
assignments, and attendance policy) - CrewControl provides the software
that records and displays this information.

2. ELIGIBILITY
You may only use this App if:
(a) you have been added as an employee by an Organization that is an
    active CrewControl customer, and
(b) you have been issued valid login credentials by that Organization.
This App is not a general-purpose consumer product and is not intended to
function, and will not function, independently of an Organization's
CrewControl account.

3. ACCURACY OF RECORDS
Attendance, location, and timesheet data recorded through the App is used
by your Organization for payroll and compliance purposes. You agree to
use the App honestly and to promptly flag any records you believe are
incorrect to your office.

4. AVAILABILITY
We aim to keep the App and its notification/push services available, but
we do not guarantee uninterrupted access. Push notifications depend on
Google's Firebase Cloud Messaging service and your device's connectivity;
delivery timing is not guaranteed, particularly if your device is offline
or powered off for an extended period.

5. LIMITATION OF LIABILITY
To the extent permitted by law, CrewControl is not liable for indirect,
incidental, or consequential damages arising from your use of the App.
Disputes about pay, attendance, or employment terms are between you and
your Organization; CrewControl's role is limited to providing the
software platform.

6. INDEMNIFICATION
You agree not to misuse the App in a way that causes loss to CrewControl
or your Organization (for example, deliberately falsifying records), and
to be responsible for any such misuse to the extent permitted by law.

7. GOVERNING LAW AND DISPUTES
These Terms are governed by the laws of [jurisdiction]. Any disputes will
first be attempted to be resolved informally; unresolved disputes will be
handled in the courts of [jurisdiction].

8. CHANGES
We may revise these Terms and Conditions periodically. We'll flag
material changes the next time you log in.

9. CONTACT
Questions about these Terms and Conditions should be directed to your
Organization's office administrator, or to [support contact / email].
''';

  // ---------------------------------------------------------------------
  static const String privacyPolicy = '''
PRIVACY POLICY
Last updated: $lastUpdated

1. WHAT DATA WE COLLECT
- Account & profile: name, phone number, photo, and other details your
  Organization enters or you update in your profile.
- Attendance & work data: check-in/check-out times, site assignments,
  leave records, and work session duration.
- Location data: your device's GPS location, collected during active
  work sessions (and, if you've enabled it, periodically in the
  background) to verify on-site attendance. Location is not collected
  when you are not checked in for work, beyond what's needed for a
  one-time on-demand location request from your office, if used.
- Pay data: salary slips, advances, and deductions, as generated by your
  Organization.
- Communications: messages sent through the in-app Chat/Inbox with your
  office.
- Device data: a Firebase Cloud Messaging (FCM) push token, used only to
  deliver notifications to your device, and basic device/app diagnostic
  information for troubleshooting.

2. HOW WE USE YOUR DATA
- To record and display your attendance, assignments, pay, and messages.
- To send you push notifications (salary slip ready, site assignment,
  new message, etc.) via Firebase Cloud Messaging.
- To let your Organization manage its workforce (this is the core
  purpose of the platform).
- We do not sell your data, and we do not use your location or attendance
  data for advertising.

3. WHO CAN SEE YOUR DATA
- Your Organization's authorized office staff can see the work-related
  data described above (attendance, location during work sessions,
  assignments, pay, messages).
- CrewControl's infrastructure providers (for example, our database host
  and Google Firebase for push notifications) process data on our behalf
  under standard confidentiality and security obligations - they do not
  independently use your data for their own purposes.
- We do not share your data with unrelated third parties or advertisers.

4. HOW LONG WE KEEP YOUR DATA
Attendance and pay records are generally kept for as long as your
Organization is required to retain them for payroll, tax, or labor-law
compliance (commonly several years after the record is created), even
after your employment ends. You can ask your Organization about its
specific retention practices.

5. YOUR CHOICES
- You can deny or later revoke location and notification permissions in
  your device settings; some features (like automatic attendance
  verification) may not work correctly if you do.
- To request a copy of, or a correction to, your personal data, contact
  your Organization's office administrator, who administers your account.

6. SECURITY
We use industry-standard measures to protect your data in transit
(HTTPS/TLS) and store authentication tokens using your device's secure
keystore (Android Keystore / iOS Keychain) rather than plain storage.

7. CHILDREN
This App is intended for use by employed adults and is not directed at
children.

8. CHANGES TO THIS POLICY
We may update this Privacy Policy from time to time. We'll flag material
changes the next time you log in.

9. CONTACT
Questions about this Privacy Policy should be directed to your
Organization's office administrator, or to [support contact / email].
''';

  // ---------------------------------------------------------------------
  static const String cookiePolicy = '''
COOKIE POLICY
Last updated: $lastUpdated

This App itself does not use browser cookies (it's a native mobile app,
not a website). This policy covers the equivalent on-device technologies
the App and its companion web dashboard use.

1. ON-DEVICE STORAGE WE USE
- Secure storage (Android Keystore / iOS Keychain) for your login session
  tokens, so you don't have to log in again every time you open the App.
- Local app storage for offline caching of your schedule, attendance
  records, and these legal documents, so the App remains usable with a
  weak or no connection.
- A Firebase Cloud Messaging (FCM) registration token, used solely to
  route push notifications to your device.

2. WHAT WE DO NOT USE
- No third-party advertising cookies or SDKs.
- No cross-app or cross-site tracking identifiers.
- No Advertising ID is read or used by this App.

3. WEB DASHBOARD
If your Organization's office staff access CrewControl through the web
dashboard, that dashboard uses standard browser cookies for session
management (keeping them logged in) and, if enabled by your Organization,
basic analytics cookies. This does not apply to the mobile App itself.

4. YOUR CONTROLS
You can clear the App's local storage at any time via your device's
Settings > Apps > CrewControl > Storage > Clear Data - this will log you
out and remove any offline-cached content, but will not affect the
records your Organization holds on the backend.

5. CHANGES
We may update this Cookie Policy from time to time. We'll flag material
changes the next time you log in.

6. CONTACT
Questions about this Cookie Policy should be directed to your
Organization's office administrator, or to [support contact / email].
''';

  // ---------------------------------------------------------------------
  static const String dataProcessingAgreement = '''
DATA PROCESSING AGREEMENT (SUMMARY)
Last updated: $lastUpdated

This summary describes the processor relationship between CrewControl
and your Organization for the personal data handled through this App.
The full Data Processing Agreement is executed between CrewControl and
your Organization as part of their commercial contract; this in-app
summary is provided to workers for transparency.

1. ROLES
- Your Organization (your employer) is the data controller: it decides
  what data is collected and why (attendance policy, pay rules, etc.).
- CrewControl is the data processor: it processes worker data only on
  your Organization's documented instructions, via the App and dashboard.

2. SCOPE OF PROCESSING
CrewControl processes the categories of data described in the Privacy
Policy (profile, attendance, location during work sessions, pay records,
in-app messages, device/notification tokens) solely to provide the
workforce management service to your Organization.

3. SUB-PROCESSORS
CrewControl uses a limited set of infrastructure sub-processors (for
example, cloud hosting and Google Firebase for push notification
delivery), each bound by confidentiality and data-protection obligations
consistent with this Agreement. CrewControl does not permit sub-processors
to use worker data for their own independent purposes.

4. SECURITY MEASURES
CrewControl maintains technical and organizational measures including
encryption in transit (TLS/HTTPS), encrypted credential storage on
device, access controls limiting dashboard access to authorized office
staff, and audit logging of legal-acceptance and account-access events.

5. DATA SUBJECT REQUESTS
Requests to access, correct, or restrict processing of your personal data
should go through your Organization's office administrator, who can
instruct CrewControl accordingly as the processor.

6. RETENTION AND DELETION
CrewControl retains data for the periods documented in the Data Retention
Policy and instructed by your Organization, and deletes or anonymizes
data once neither party is required to retain it.

7. INTERNATIONAL TRANSFERS
Where data is processed in a country other than your Organization's, this
is done under appropriate safeguards consistent with applicable data
protection law.

8. CONTACT
Questions about data processing should be directed to your Organization's
office administrator, or to [support contact / email].
''';

  // ---------------------------------------------------------------------
  static const String securityPrivacyStatement = '''
SECURITY & PRIVACY STATEMENT
Last updated: $lastUpdated

This statement summarizes, in plain language, the concrete security
measures protecting your data in this App.

1. DATA IN TRANSIT
All communication between the App and our servers uses HTTPS/TLS
encryption. No login credentials, tokens, or work data are ever sent
unencrypted.

2. DATA AT REST ON YOUR DEVICE
- Authentication tokens are stored in your device's secure hardware-backed
  keystore (Android Keystore / iOS Keychain), not in plain app storage.
- We do not store your password on the device after login.

3. AUTHENTICATION
- Sessions expire automatically and are refreshed using a short-lived
  token exchange; a revoked or expired session requires signing in again.
- Repeated failed logins may be rate-limited by your Organization's
  administrator settings.

4. LOCATION DATA HANDLING
Location is requested only when needed for attendance verification,
explained in-app before the permission prompt appears, and is not
collected continuously in the background beyond what your Organization
has configured for legitimate attendance/geofence purposes.

5. ACCESS CONTROLS
Only your Organization's authorized office staff can view your work data
through the dashboard; role-based permissions restrict what each staff
account can see or edit.

6. AUDIT LOGGING
Security-relevant events - including legal-document acceptance, login,
and account-deactivation requests - are logged with a timestamp for audit
and compliance purposes.

7. INCIDENT RESPONSE
In the event of a data security incident affecting your personal data, we
will notify your Organization, who is responsible for notifying affected
workers as required by applicable law.

8. REPORTING A CONCERN
If you believe you've found a security vulnerability, report it to
[security contact / email] rather than to your Organization's office
staff, so it can be triaged directly by the engineering team.

9. CONTACT
Questions about this statement should be directed to your Organization's
office administrator, or to [support contact / email].
''';

  // ---------------------------------------------------------------------
  static const String dataRetentionPolicy = '''
DATA RETENTION POLICY
Last updated: $lastUpdated

1. WHY WE RETAIN DATA
As a workforce management platform, CrewControl retains certain worker
data on behalf of your Organization because your Organization is legally
required to keep payroll, attendance, and tax-related records for a
minimum period under applicable labor law, even after your employment
ends.

2. RETENTION PERIODS
- Attendance & timesheet records: retained for the period your
  Organization configures to meet its statutory record-keeping
  obligations (commonly several years).
- Payroll & salary slip records: retained per your Organization's payroll
  and tax retention requirements.
- Chat/messaging history: retained according to your Organization's
  configured retention policy.
- Legal-acceptance audit records (who accepted which version, when):
  retained indefinitely as a compliance record, since it evidences
  consent to terms that were in effect at a given time.
- Location data tied to a specific attendance event: retained alongside
  that attendance record for the same period; location is not retained
  independently of an attendance/work event.

3. WHAT HAPPENS WHEN YOU LEAVE YOUR ORGANIZATION
Your login is deactivated and cannot be used to sign in, but your work
records (described above) are NOT deleted, because they belong to your
Organization and are needed for payroll, audit, and legal-compliance
purposes. See "Request Account Deactivation" in the App's Legal & Privacy
settings for how deactivation works.

4. DELETION
Data is deleted or anonymized once your Organization confirms it is no
longer required to retain it under its own record-retention schedule and
applicable law. Because these records are the property and legal
obligation of your Organization (not CrewControl unilaterally), deletion
requests are handled through your Organization's administrator.

5. CHANGES
We may update this Data Retention Policy from time to time. We'll flag
material changes the next time you log in.

6. CONTACT
Questions about data retention should be directed to your Organization's
office administrator, or to [support contact / email].
''';

  /// All documents a worker must review and accept, in display order.
  /// UI (login checkbox, Legal & Privacy settings page) should iterate this
  /// list rather than hardcoding individual documents, so adding a new
  /// required document only requires editing this file.
  static const List<LegalDocument> all = [
    LegalDocument(
      key: 'privacy_policy',
      title: 'Privacy Policy',
      version: currentAcceptanceVersion,
      effectiveDate: effectiveDate,
      content: privacyPolicy,
    ),
    LegalDocument(
      key: 'terms_and_conditions',
      title: 'Terms & Conditions',
      version: currentAcceptanceVersion,
      effectiveDate: effectiveDate,
      content: termsAndConditions,
    ),
    LegalDocument(
      key: 'terms_of_use',
      title: 'Terms of Use',
      version: currentAcceptanceVersion,
      effectiveDate: effectiveDate,
      content: termsOfUse,
    ),
    LegalDocument(
      key: 'cookie_policy',
      title: 'Cookie Policy',
      version: currentAcceptanceVersion,
      effectiveDate: effectiveDate,
      content: cookiePolicy,
    ),
    LegalDocument(
      key: 'data_processing_agreement',
      title: 'Data Processing Agreement',
      version: currentAcceptanceVersion,
      effectiveDate: effectiveDate,
      content: dataProcessingAgreement,
    ),
    LegalDocument(
      key: 'security_privacy_statement',
      title: 'Security & Privacy Statement',
      version: currentAcceptanceVersion,
      effectiveDate: effectiveDate,
      content: securityPrivacyStatement,
    ),
    LegalDocument(
      key: 'data_retention_policy',
      title: 'Data Retention Policy',
      version: currentAcceptanceVersion,
      effectiveDate: effectiveDate,
      content: dataRetentionPolicy,
    ),
  ];
}

File: backend/src/config/legalDocuments.js
// backend/src/config/legalDocuments.js
//
// Single source of truth for every legal document CrewControl shows users
// (signup consent, Account & Security > Legal & Privacy). Content lives here
// as structured data (not markdown/html files) so both the "accept on
// signup" flow and the "view in Account & Security" flow render identical,
// versioned content from one place.
//
// LEGAL_BUNDLE_VERSION is a single version for the *set* of documents the
// user consents to as one action ("I agree to the Privacy Policy, T&Cs,
// ..."). Bump it (and EFFECTIVE_DATE/LAST_UPDATED) whenever the substance of
// any document below changes - every user will then be required to
// re-review and re-accept on next login (see legal.controller.js).
//
// Fill in the placeholders (marked {{LIKE_THIS}}) with real business/legal
// details before shipping to production.

export const LEGAL_BUNDLE_VERSION = '1.0.0';
export const LEGAL_EFFECTIVE_DATE = '2026-07-19';
export const LEGAL_LAST_UPDATED = '2026-07-19';

const PLACEHOLDERS = {
  COMPANY_NAME: '{{COMPANY_LEGAL_NAME}}',
  BRAND_NAME: 'CrewControl',
  REGISTERED_ADDRESS: '{{REGISTERED_BUSINESS_ADDRESS}}',
  JURISDICTION: '{{GOVERNING_JURISDICTION, e.g. Dubai, United Arab Emirates}}',
  LEGAL_EMAIL: '{{LEGAL_CONTACT_EMAIL}}',
  PRIVACY_EMAIL: '{{PRIVACY_CONTACT_EMAIL}}',
  DPO_EMAIL: '{{DATA_PROTECTION_OFFICER_EMAIL}}',
  SUPPORT_EMAIL: '{{SUPPORT_EMAIL}}',
};

// Reusable section shape: { heading, body: [ { type: 'p'|'ul', text|items } ] }
const p = (text) => ({ type: 'p', text });
const ul = (items) => ({ type: 'ul', items });

const meta = (overrides = {}) => ({
  version: LEGAL_BUNDLE_VERSION,
  effectiveDate: LEGAL_EFFECTIVE_DATE,
  lastUpdated: LEGAL_LAST_UPDATED,
  ...overrides,
});

export const LEGAL_DOCUMENTS = {
  'privacy-policy': {
    slug: 'privacy-policy',
    title: 'Privacy Policy',
    ...meta(),
    sections: [
      {
        heading: '1. Introduction',
        body: [
          p(`This Privacy Policy explains how ${PLACEHOLDERS.COMPANY_NAME} ("${PLACEHOLDERS.BRAND_NAME}", "we", "us", "our") collects, uses, stores, shares, and protects information when you use our workforce management and AI-driven invoicing platform (the "Service"), whether via our web dashboard or companion mobile application.`),
          p(`We are committed to processing personal data lawfully, fairly, and transparently, in line with the UAE Personal Data Protection Law (Federal Decree-Law No. 45 of 2021, "PDPL"), the EU General Data Protection Regulation ("GDPR") where applicable, and internationally recognized SaaS data-protection practices.`),
        ],
      },
      {
        heading: '2. Information We Collect',
        body: [
          p('We collect the following categories of information:'),
          ul([
            'Account information: name, email address, password (hashed), mobile number, role, and company affiliation.',
            'Company information: legal name, tax registration number, registered address, contact details, and branding assets (logo, stamp, authorized signature).',
            'Employee records: names, contact details, trades/roles, employment dates, and employment documents you upload.',
            'Attendance and time data: check-in/check-out timestamps, work sessions, leave records.',
            'Location (GPS) data: device location captured during active shifts to verify on-site attendance, where enabled.',
            'Timesheets and uploaded documents: PDF timesheets, invoices, and related files you upload for processing.',
            'AI-extracted invoice data: structured data (line items, amounts, dates, vendor details) produced by our AI/OCR pipeline from uploaded documents.',
            'Generated invoices and payroll information: invoices, salary slips, deductions, and payment-related records you create in the Service.',
            'Chat messages: messages exchanged between office/admin users and employees within the Service.',
            'Notifications: in-app and push notification content and delivery metadata.',
            'Analytics and usage data: feature usage, session activity, device/browser type, and IP address.',
            'Audit logs: records of security-relevant actions (logins, permission changes, data exports, legal-agreement acceptances) kept for compliance purposes.',
          ]),
        ],
      },
      {
        heading: '3. Purpose of Processing',
        body: [
          p('We process personal data to:'),
          ul([
            'Provide, operate, and maintain the Service, including attendance tracking, invoicing, and payroll features.',
            'Verify identity and authenticate accounts.',
            'Process uploaded documents using AI/OCR to extract structured invoice and timesheet data.',
            'Generate invoices, salary slips, and expense reports on your behalf.',
            'Send transactional notifications (e.g. new chat messages, attendance alerts) and, where you opt in, product updates.',
            'Monitor, secure, and improve the Service, including fraud prevention and abuse detection.',
            'Comply with legal, tax, and regulatory obligations.',
          ]),
        ],
      },
      {
        heading: '4. Storage and Encryption',
        body: [
          p('Data is stored on infrastructure operated by our hosting and database providers, with access restricted to authorized personnel and systems. Data in transit is encrypted using TLS. Sensitive fields (such as password credentials) are stored using one-way cryptographic hashing and are never stored or transmitted in plaintext. Uploaded files and generated documents are stored in access-controlled object storage.'),
        ],
      },
      {
        heading: '5. AI Processing',
        body: [
          p('Uploaded timesheets and invoice documents may be processed by our AI/OCR pipeline (which may include third-party AI or OCR providers acting as sub-processors) to extract structured data automatically. This processing is performed to deliver the core invoicing functionality of the Service. We do not use your uploaded business documents to train AI models for use outside your account.'),
        ],
      },
      {
        heading: '6. Cookies and Analytics',
        body: [
          p('We use essential, authentication, session, and (where enabled) analytics cookies. See our Cookie Policy for full details on the categories of cookies used and how to manage your preferences.'),
        ],
      },
      {
        heading: '7. Location Tracking',
        body: [
          p('Where the mobile companion app is used for attendance, we collect device GPS location at check-in/check-out and, where configured by the employer, during active shifts, solely to verify on-site presence. Location tracking can be controlled at the device level via mobile OS permissions.'),
        ],
      },
      {
        heading: '8. Your Rights',
        body: [
          p('Subject to applicable law (including UAE PDPL and, where applicable, GDPR), you may have the right to:'),
          ul([
            'Access the personal data we hold about you.',
            'Request correction of inaccurate or incomplete data.',
            'Request deletion of your personal data ("right to erasure"), subject to legal retention obligations.',
            'Request a portable copy of your data in a structured, machine-readable format.',
            'Object to or restrict certain processing activities.',
            'Withdraw consent where processing is based on consent, without affecting processing carried out before withdrawal.',
          ]),
          p(`To exercise these rights, contact us at ${PLACEHOLDERS.PRIVACY_EMAIL}.`),
        ],
      },
      {
        heading: '9. Data Deletion and Portability',
        body: [
          p('You may request deletion of your account and associated personal data at any time, subject to retention periods described in our Data Retention Policy (e.g. for tax, audit, and dispute-resolution purposes). Where technically feasible, we will provide your data in a portable format (such as CSV or JSON) upon request.'),
        ],
      },
      {
        heading: '10. International Transfers',
        body: [
          p(`Where personal data is transferred outside the ${PLACEHOLDERS.JURISDICTION} (for example, to cloud infrastructure or sub-processors located elsewhere), we take reasonable steps to ensure such transfers are subject to appropriate safeguards, such as standard contractual clauses or equivalent mechanisms recognized under UAE PDPL and GDPR.`),
        ],
      },
      {
        heading: '11. Third-Party Processors',
        body: [
          p('We engage vetted third-party service providers (sub-processors) to host infrastructure, send transactional email/SMS/push notifications, process payments, and perform AI/OCR extraction. Each sub-processor is bound by contractual confidentiality and data-protection obligations consistent with this Policy. A current list of sub-processor categories is available on request.'),
        ],
      },
      {
        heading: '12. Contact Us',
        body: [
          p(`Questions about this Privacy Policy or our data practices can be directed to ${PLACEHOLDERS.PRIVACY_EMAIL} or our Data Protection contact at ${PLACEHOLDERS.DPO_EMAIL}.`),
        ],
      },
    ],
  },

  'terms-and-conditions': {
    slug: 'terms-and-conditions',
    title: 'Terms & Conditions',
    ...meta(),
    sections: [
      {
        heading: '1. Account Ownership',
        body: [
          p(`The individual who registers for the Service is the "Account Owner" and is responsible for all activity under that account, including activity by employees or team members granted access. You must provide accurate registration information and keep it up to date.`),
        ],
      },
      {
        heading: '2. Subscription and Billing',
        body: [
          p('Access to paid features requires an active subscription. Subscription fees are billed in advance on a recurring basis (monthly or yearly, as selected) via our payment processor. Prices may change with reasonable prior notice.'),
        ],
      },
      {
        heading: '3. Cancellation and Refunds',
        body: [
          p('You may cancel your subscription at any time; access continues until the end of the current billing period. Except where required by applicable law, fees already paid are non-refundable. Cancellation does not entitle you to a pro-rated refund for unused time unless otherwise stated at the time of purchase.'),
        ],
      },
      {
        heading: '4. Responsibilities',
        body: [
          p('You are responsible for the accuracy of data entered into the Service (including attendance, invoice, and payroll data) and for maintaining the confidentiality of your login credentials.'),
        ],
      },
      {
        heading: '5. Service Availability',
        body: [
          p('We aim to keep the Service available at all times but do not guarantee uninterrupted access. Scheduled maintenance, third-party outages, or force majeure events may result in temporary unavailability.'),
        ],
      },
      {
        heading: '6. Intellectual Property',
        body: [
          p(`All rights, title, and interest in the Service, including software, design, and branding, remain the exclusive property of ${PLACEHOLDERS.COMPANY_NAME} and its licensors. You retain ownership of the business data you upload or generate through the Service.`),
        ],
      },
      {
        heading: '7. Limitation of Liability',
        body: [
          p(`To the maximum extent permitted by law, ${PLACEHOLDERS.COMPANY_NAME} shall not be liable for indirect, incidental, special, or consequential damages arising from use of the Service, including reliance on AI-extracted data, which should be reviewed for accuracy before use in official filings.`),
        ],
      },
      {
        heading: '8. Indemnification',
        body: [
          p(`You agree to indemnify and hold harmless ${PLACEHOLDERS.COMPANY_NAME} from claims arising out of your misuse of the Service or violation of these Terms.`),
        ],
      },
      {
        heading: '9. Force Majeure',
        body: [
          p('Neither party is liable for delays or failures caused by events beyond reasonable control, including natural disasters, internet or infrastructure outages, or governmental action.'),
        ],
      },
      {
        heading: '10. Governing Law and Dispute Resolution',
        body: [
          p(`These Terms are governed by the laws of ${PLACEHOLDERS.JURISDICTION}. Any dispute arising from these Terms will first be addressed through good-faith negotiation, and if unresolved, submitted to the competent courts or arbitration body of ${PLACEHOLDERS.JURISDICTION}.`),
        ],
      },
      {
        heading: '11. Suspension and Termination',
        body: [
          p('We may suspend or terminate accounts that violate these Terms, our Terms of Use, or applicable law, or that pose a security risk to the Service or other users. Where reasonably possible, we will provide notice before suspension.'),
        ],
      },
    ],
  },

  'terms-of-use': {
    slug: 'terms-of-use',
    title: 'Terms of Use',
    ...meta(),
    sections: [
      {
        heading: '1. Acceptable Use',
        body: [
          p('The Service is provided for legitimate workforce management and invoicing purposes. You agree to use it only as intended and in compliance with applicable law.'),
        ],
      },
      {
        heading: '2. Prohibited Activities',
        body: [
          p('You must not:'),
          ul([
            'Share your login credentials with unauthorized individuals or allow account access outside your organization.',
            'Attempt to gain unauthorized access to any part of the Service, other accounts, or underlying infrastructure ("hacking").',
            'Reverse engineer, decompile, or disassemble any part of the Service.',
            'Scrape, crawl, or bulk-extract data from the Service outside of provided export features.',
            'Send spam, unsolicited messages, or use the chat feature for unrelated marketing.',
            'Harass, abuse, or threaten other users through the Service.',
            'Upload or distribute malware, viruses, or other harmful code.',
            'Use the Service for any illegal activity, including fraud, money laundering, or falsification of official documents.',
          ]),
        ],
      },
      {
        heading: '3. Consequences of Violation',
        body: [
          p('Violation of this Terms of Use may result in suspension or termination of your account, and where applicable, referral to law enforcement.'),
        ],
      },
    ],
  },

  'cookie-policy': {
    slug: 'cookie-policy',
    title: 'Cookie Policy',
    ...meta(),
    sections: [
      {
        heading: '1. What Are Cookies',
        body: [
          p('Cookies are small text files stored on your device that help us operate and improve the Service.'),
        ],
      },
      {
        heading: '2. Categories of Cookies We Use',
        body: [
          ul([
            'Essential cookies: required for core functionality such as page routing and security; cannot be disabled.',
            'Authentication cookies: keep you securely signed in between requests.',
            'Session cookies: maintain your session state while you use the Service and expire when you close your browser or log out.',
            'Preference cookies: remember settings such as language or display preferences.',
            'Analytics cookies: help us understand feature usage and improve the Service (used only where enabled).',
          ]),
        ],
      },
      {
        heading: '3. Managing Cookies',
        body: [
          p('You can manage or disable non-essential cookies through your browser settings. Disabling essential cookies may prevent parts of the Service from functioning correctly.'),
        ],
      },
    ],
  },

  'data-processing-agreement': {
    slug: 'data-processing-agreement',
    title: 'Data Processing Agreement (DPA)',
    ...meta(),
    sections: [
      {
        heading: '1. Roles',
        body: [
          p(`For personal data of your employees processed through the Service (e.g. attendance, payroll), you act as the "Controller" and ${PLACEHOLDERS.COMPANY_NAME} acts as the "Processor" under UAE PDPL and, where applicable, GDPR.`),
        ],
      },
      {
        heading: '2. Sub-processors',
        body: [
          p('We may engage sub-processors (e.g. cloud hosting, email/SMS delivery, AI/OCR providers) to perform processing on our behalf. Sub-processors are bound by data-protection obligations no less protective than those in this DPA. We will notify you of material changes to our sub-processor list where required by law or contract.'),
        ],
      },
      {
        heading: '3. Security Measures',
        body: [
          p('We implement appropriate technical and organizational measures, including encryption in transit, password hashing, role-based access control, and audit logging, as further described in our Security & Privacy Statement.'),
        ],
      },
      {
        heading: '4. Breach Notification',
        body: [
          p('In the event of a personal data breach affecting your data, we will notify you without undue delay after becoming aware of it, providing information reasonably necessary for you to meet your own regulatory notification obligations.'),
        ],
      },
      {
        heading: '5. Retention and Deletion',
        body: [
          p('Personal data is retained in line with our Data Retention Policy and deleted or anonymized upon expiry of the applicable retention period, or upon valid deletion request, subject to legal holds.'),
        ],
      },
      {
        heading: '6. Cross-Border Transfers',
        body: [
          p('Any cross-border transfer of personal data is carried out subject to appropriate safeguards consistent with UAE PDPL and GDPR requirements.'),
        ],
      },
      {
        heading: '7. Audit Rights',
        body: [
          p('Upon reasonable request and subject to confidentiality obligations, we will provide information reasonably necessary to demonstrate compliance with this DPA, including relevant security documentation.'),
        ],
      },
    ],
  },

  'security-privacy-statement': {
    slug: 'security-privacy-statement',
    title: 'Security & Privacy Statement',
    ...meta(),
    sections: [
      {
        heading: '1. Encryption',
        body: [
          p('Data in transit is encrypted using TLS. Sensitive data at rest is protected using industry-standard encryption provided by our infrastructure and database providers.'),
        ],
      },
      {
        heading: '2. Password Hashing',
        body: [
          p('Passwords are never stored in plaintext. We use one-way, salted cryptographic hashing (bcrypt) to store password credentials.'),
        ],
      },
      {
        heading: '3. Authentication',
        body: [
          p('The Service supports token-based authentication and optional two-factor authentication (TOTP) for an additional layer of account security.'),
        ],
      },
      {
        heading: '4. Role-Based Access Control',
        body: [
          p('Access to features and data within an organization\u2019s account is scoped by role (e.g. owner/admin vs. employee), limiting each user to the data and actions relevant to their role.'),
        ],
      },
      {
        heading: '5. Audit Logs',
        body: [
          p('Security-relevant events, including logins, legal-agreement acceptances, and sensitive data changes, are recorded in audit logs for accountability and incident investigation.'),
        ],
      },
      {
        heading: '6. Backups and Monitoring',
        body: [
          p('Production data is backed up on a regular schedule, and infrastructure is monitored for availability and anomalous activity.'),
        ],
      },
      {
        heading: '7. Incident Response',
        body: [
          p('We maintain an internal process for identifying, containing, and remediating security incidents, including notification obligations described in our Data Processing Agreement.'),
        ],
      },
      {
        heading: '8. Infrastructure and Tenant Isolation',
        body: [
          p('Each company/organization\u2019s data is logically isolated within the Service, so that one tenant cannot access another tenant\u2019s data.'),
        ],
      },
      {
        heading: '9. AI Security',
        body: [
          p('Documents processed through our AI/OCR pipeline are handled within access-controlled processing environments and are not used to train models for use outside your account.'),
        ],
      },
      {
        heading: '10. Secure Software Development Lifecycle',
        body: [
          p('Changes to the Service go through code review and testing practices intended to identify and remediate security issues before release.'),
        ],
      },
    ],
  },

  'data-retention-policy': {
    slug: 'data-retention-policy',
    title: 'Data Retention Policy',
    ...meta(),
    sections: [
      {
        heading: '1. Purpose',
        body: [
          p('This policy describes how long we retain different categories of data and the basis for those retention periods.'),
        ],
      },
      {
        heading: '2. Retention Periods',
        body: [
          ul([
            'Account data: retained for the life of the account, and up to 30 days after account deletion to allow for recovery, then permanently deleted or anonymized.',
            'Attendance records: retained for up to 7 years to support payroll, tax, and labor-law compliance.',
            'Payroll information: retained for up to 7 years in line with common tax and labor recordkeeping requirements.',
            'Invoices and AI-extracted invoice data: retained for up to 7 years for tax and audit purposes.',
            'Uploaded documents (timesheets, templates, signatures): retained while the account is active and for the same period as the records they support.',
            'Chat messages: retained for up to 2 years, or as long as the underlying employment relationship is active, whichever is longer.',
            'Notifications: retained for up to 12 months.',
            'Audit logs: retained for up to 7 years to support security investigations and compliance obligations.',
            'Deleted accounts: personal data is deleted or anonymized within 30 days of a confirmed deletion request, except where retention is legally required.',
            'Backups: rolled over on a regular cycle; deleted data is purged from backups within a reasonable period after the standard backup retention window.',
            'Temporary files (e.g. AI processing intermediates): deleted automatically, typically within 30 days of processing.',
          ]),
        ],
      },
      {
        heading: '3. Legal Holds',
        body: [
          p('Where data is subject to a legal hold, dispute, or regulatory investigation, retention periods above may be extended until the matter is resolved.'),
        ],
      },
    ],
  },
};

export const LEGAL_DOCUMENT_LIST = Object.values(LEGAL_DOCUMENTS).map((doc) => ({
  slug: doc.slug,
  title: doc.title,
  version: doc.version,
  effectiveDate: doc.effectiveDate,
  lastUpdated: doc.lastUpdated,
}));

export const getLegalDocument = (slug) => LEGAL_DOCUMENTS[slug] || null;

export const LEGAL_CONSENT_LABEL =
  'I have read and agree to the Privacy Policy, Terms & Conditions, Terms of Use, Cookie Policy, ' +
  'Data Processing Agreement, Security & Privacy Policy, and Consent to Data Processing.';

export default {
  LEGAL_BUNDLE_VERSION,
  LEGAL_EFFECTIVE_DATE,
  LEGAL_LAST_UPDATED,
  LEGAL_DOCUMENTS,
  LEGAL_DOCUMENT_LIST,
  getLegalDocument,
  LEGAL_CONSENT_LABEL,
};
